tests/react/useAtomValue.test.tsx

```diff
@@ -30,3 +30,110 @@ it('useAtomValue basic test', async () => {
   await userEvent.click(screen.getByText('dispatch'))
   await screen.findByText('count: 1')
 })
+
+it('useAtomValue with async atom (promise)', async () => {
+  const asyncAtom = atom(async () => 42)
+
+  const AsyncComponent = () => {
+    const value = useAtomValue(asyncAtom)
+
+    return <div>value: {value}</div>
+  }
+
+  await act(async () => {
+    render(
+      <StrictMode>
+        <Suspense fallback="loading">
+          <AsyncComponent />
+        </Suspense>
+      </StrictMode>,
+    )
+  })
+
+  await screen.findByText('value: 42')
+})
+
+class ErrorBoundary extends React.Component<
+  { children: React.ReactNode },
+  { error: Error | null }
+> {
+  constructor(props: { children: React.ReactNode }) {
+    super(props)
+
+    this.state = { error: null }
+  }
+
+  static getDerivedStateFromError(error: Error) {
+    return { error }
+  }
+
+  render() {
+    if (this.state.error) {
+      return <div>error: {this.state.error.message}</div>
+    }
+
+    return this.props.children
+  }
+}
+
+it('useAtomValue with error throwing atom', async () => {
+  const errorAtom = atom(() => {
+    throw new Error('fail')
+  })
+
+  const ErrorComponent = () => {
+    useAtomValue(errorAtom)
+
+    return <div>no error</div>
+  }
+
+  render(
+    <StrictMode>
+      <ErrorBoundary>
+        <ErrorComponent />
+      </ErrorBoundary>
+    </StrictMode>,
+  )
+
+  await screen.findByText('error: fail')
+})
+
+it('useAtomValue with atom returning object', async () => {
+  const objAtom = atom({ a: 1, b: 2 })
+
+  const ObjComponent = () => {
+    const value = useAtomValue(objAtom)
+
+    return (
+      <div>
+        obj: {value.a},{value.b}
+      </div>
+    )
+  }
+
+  render(
+    <StrictMode>
+      <ObjComponent />
+    </StrictMode>,
+  )
+
+  await screen.findByText('obj: 1,2')
+})
+
+it('useAtomValue with atom returning function', async () => {
```

Comment:
This usage isn't very recommended because it's easy to violate React rules. To avoid misconception, can we remove this test or add caveat comment?

Reply:
Sure! Just removed this!